api/versions.py

```
@@ -0,0 +1,97 @@
+logger = logging.getLogger(__name__)
+
+class RecordVersionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = RecordVersion
+        fields = ("version", "prev_version", "changed_at", "diff", "meta")
+
+
+class VersionViewSet(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+    
+    def list(self, request, resource_type: str | None = None, resource_id: str | None = None) -> Response:
```

Ruff (0.12.2)

Missing type annotation for function argument `request`

(ANN001)